lib/open_api_spex/cast.ex

```diff
@@ -100,10 +100,17 @@ defmodule OpenApiSpex.Cast do
 
   """
 
-  @spec cast(schema_or_reference | nil, term(), map(), [cast_opt()]) ::
+  @spec cast(schema_or_reference | nil, term(), map(), read_write_scope | nil, [cast_opt()]) ::
           {:ok, term()} | {:error, [Error.t()]}
-  def cast(schema, value, schemas \\ %{}, opts \\ []) do
-    ctx = %__MODULE__{schema: schema, value: value, schemas: schemas, opts: opts}
+  def cast(schema, value, schemas \\ %{}, read_write_scope \\ nil, opts \\ []) do
```

Comment:
Would adding a default parameter in the 4th position here break existing code that was supplying `schema`, `value`, `schemas` and `opts` ?

Maybe the `read_write_scope` value can be passed in `opts`, and `Keyword.pop` it out to place in the struct field ?

Reply:
Alternatively we can update the call sites in `operation2.ex` and `cast_parameters.ex` to use `cast/1` with the `%Cast{}` struct built inline with the `read_write_scope` field set.

Reply:
I went with the later option. I thought this might be fine since the only two calls to `Cast.cast/4` were actually the ones needing the `:write` option, but it will indeed definitely  break any external code.